### PR TITLE
feat: Add `define‑properties`

### PR DESCRIPTION
**Pull request by ExE-Boss**
Please fill in this template.

- [x] Use a meaningful title for the pull request. Include the name of the package modified.
- [x] Test the change in your own code. (Compile and run.)
- [x] Add or edit tests to reflect the change. (Run with `npm test`.)
- [x] Follow the advice from the [readme](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#make-a-pull-request).
- [x] Avoid [common mistakes](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#common-mistakes).
- [x] Run `npm run lint package-name` (or `tsc` if no `tslint.json` is present).

If adding a new definition:
- [x] The package does not already provide its own types, or cannot have its `.d.ts` files generated via `--declaration`
- [x] If this is for an NPM package, match the name. If not, do not conflict with the name of an NPM package.
- [x] Create it with `dts-gen --dt`, not by basing it on an existing project.
- [x] Represents shape of module/library [correctly](https://www.typescriptlang.org/docs/handbook/declaration-files/library-structures.html)
- [x] `tslint.json` should be present and it shouldn't have any additional or disabling of rules. Just content as `{ "extends": "dtslint/dt.json" }`. If for reason the some rule need to be disabled, disable it for that line using `// tslint:disable-next-line [ruleName]`  and not for whole package so that the need for disabling can be reviewed.
- [x] `tsconfig.json` should have `noImplicitAny`, `noImplicitThis`, `strictNullChecks`, and `strictFunctionTypes` set to `true`.

---

review?(@ljharb, @andrewbranch)

### CODE REVIEW

File: types/define-properties/index.d.ts [hunk 1]
```diff
@@ -0,0 +1,27 @@
...
+
+/**
+ * Defines new properties in `map` as non-enumerable if they don't already
+ * exist on `object`.
+ *
+ * @param object The object to define non-enumerable properties on.
+ * @param map The map of newly defined properties.
+ * @param predicates The optional predicates map, return `true` to override existing properties on `object`.
+ */
+declare function defineProperties<K extends keyof any>(
```
Comment (ljharb) [inline types/define-properties/index.d.ts]: why is this any? The K must be a Property Key, ie a string/number/symbol/bigint

**Review by ljharb — APPROVED** [review]

File: types/define-properties/index.d.ts (hunk 1 above)
Comment (ExE-Boss) [reply types/define-properties/index.d.ts]: It extends `keyof any`, which resolves to `string | number | symbol`.

Also, last I checked, `bigint` couldn’t be used to index properties directly in TypeScript (unless they changed that).

Comment (ljharb) [reply types/define-properties/index.d.ts]: true, that hasn’t been fixed yet.

File: types/define-properties/.editorconfig [hunk 2]
```diff
@@ -0,0 +1,3 @@
+# This package uses tabs
+[*]
+indent_style = tab
```
Comment (andrewbranch) [inline types/define-properties/.editorconfig]: Is this necessary?

Comment (ExE-Boss) [reply types/define-properties/.editorconfig]: Yes.

Otherwise, Prettier will&nbsp;convert tabs&nbsp;to&nbsp;spaces, which&nbsp;I&nbsp;don’t&nbsp;want.

**Pull request merged**